azkaban-common/src/main/java/azkaban/imagemgmt/daos/HPFlowDaoImpl.java

```diff
@@ -0,0 +1,169 @@
+
+  // Flow ID is a fully qualified flow name with format,
+  // <project_name>.<flow_ame>
+  private static String INSERT_HP_FLOW_OWNER =
+      "INSERT into hp_flow_owners (name, created_on, created_by, "
+      + "modified_on, modified_by ) "
+      + "values (?, ?, ?, ?, ?)";
+
+  private static String INSERT_HP_FLOWS =
+      "INSERT into hp_flows (flow_id, created_on, created_by) "
+      + "values (?, ?, ?)";
+
+  @Inject
+  public HPFlowDaoImpl(final DatabaseOperator databaseOperator) {
+    this.databaseOperator = databaseOperator;
+  }
+
+  /**
+   * Returns if the provided user has high priority flow management access.
+   * @param userId : owner of high priority flows.
+   * @return
+   */
+  @Override
+  public boolean isHPFlowOwner(final String userId) {
+    final FetchHPFlowOwnership fetchHPFlowOwnership = new FetchHPFlowOwnership();
+    try {
+      final String returnedName = this.databaseOperator
+          .query(FetchHPFlowOwnership.FETCH_HP_OWNER_BY_NAME,
+              fetchHPFlowOwnership, userId.toLowerCase());
+      log.info(String.format(
+          "HPFlowDao : Returned user id %s for given user id %s", returnedName, userId));
+      return returnedName.equalsIgnoreCase(userId);
+    } catch (final SQLException e) {
+      log.error(FetchHPFlowOwnership.FETCH_HP_OWNER_BY_NAME + " failed.", e);
+      throw new ImageMgmtDaoException(ErrorCode.INTERNAL_SERVER_ERROR, String.format(
+          "Unable to fetch HP flow ownership for %s.", userId));
+    }
+  }
+
+  /**
+   * Adds a list of flows into high priority flows db.
+   * @param flowIds : list of flows.
+   * @param userId : userId of owner who is adding the HP flows.
+   * @return
+   */
+  @Override
+  public int addHPFlows(final List<String> flowIds, final String userId) {
+    // Create the transaction block
+    final SQLTransaction<Integer> insertAllFlows = transOperator -> {
+      // Insert each flow in its own row with same userId and created time.
+      final Timestamp currTimeStamp = Timestamp.valueOf(LocalDateTime.now());
+      int count = 0;
+      for (final String flowId : flowIds) {
+        transOperator.update(INSERT_HP_FLOWS, flowId, currTimeStamp, userId);
+        count++;
+      }
+      transOperator.getConnection().commit();
+      return count;
+    };
+
+    int count = 0;
+    try {
+      count = this.databaseOperator.transaction(insertAllFlows);
+      if (count < 1) {
+        log.error("Exception while adding HP flows");
+        throw new ImageMgmtDaoException(ErrorCode.BAD_REQUEST,
+            "Exception while adding HP flows");
+      }
+      log.info("Added the HP flows by user " + userId);
+      return count;
+    } catch (final SQLException e) {
+      log.error("Unable to add HP flows", e);
+      String errorMessage = "";
+      if (e.getErrorCode() == SQL_ERROR_CODE_DUPLICATE_ENTRY) {
+        errorMessage = "Reason: One or more flows already exists.";
+      } else {
+        errorMessage = "Reason: ";
```

Comment:
Shall we say "Reason: Unknown" here, or something else?

Reply:
Will do.